vita3k/shader/src/translator/ialu.cpp

```diff
@@ -121,10 +115,27 @@ bool USSETranslatorVisitor::vbw(
     case Opcode::SHR: operation = spv::Op::OpShiftRightLogical; break;
     default: return false;
     }
-
-    result = m_b.createBinOp(operation, type_ui32, src1, src2);
-    if (m_b.isFloatType(m_b.getTypeId(src2))) {
-        result = m_b.createUnaryOp(spv::Op::OpBitcast, type_f32, src2);
+    //optimisation. (any OR 0 || any XOR 0 || any AND 0xFFFFFFFF) -> assign
+    bool is_no_op = (operation == spv::Op::OpBitwiseOr || operation == spv::Op::OpBitwiseXor) && m_b.getOpCode(src2) == spv::Op::OpConstant && m_b.getConstantScalar(src2) == 0;
+    is_no_op = is_no_op || (operation == spv::Op::OpBitwiseAnd) && m_b.getOpCode(src2) == spv::Op::OpConstant && m_b.getConstantScalar(src2) == 0xFFFFFFFF;
```

Comment:
```suggestion
    is_no_op = is_no_op || (operation == spv::Op::OpBitwiseAnd) && (m_b.getOpCode(src2) == spv::Op::OpConstant) && (m_b.getConstantScalar(src2) == 0xFFFFFFFF);
```

For a bit readability. Maybe compiler can be confused without brackets

Reply:
Tnx. I've thought that such brackets around comparisons was a pascal code style, not used in C++ , but I like this (cause I was Pascal and Delphi programmer for many years)